_pages/home/updating-to-version-2211.35/modernize-apps-migrated-from-6.8-to-2211.19.md

```diff
@@ -1,466 +1,473 @@
 
-```bash
-rm tsconfig.server.json
-```
+   For example, you can do this by running the following command on Mac or Linux:
 
-### `src/app.server.module.ts`
+   ```bash
+   mv src/app/app.server.module.ts src/app/app.module.server.ts
+   ```
 
-Rename file from `app.server.module.ts` to `app.module.server.ts` (i.e. swap the words `server` and `module`).
+1. In `src/main.server.ts`, change the export path of the `AppServerModule` from `./app/app.server.module'` to `./app/app.module.server'`.
+1. And `as default` to the `AppServerModule` export, as shown in the following example:
 
-Example command on Mac/Linux:
+   ```ts
+   export { AppServerModule as default } from './app/app.module.server';
+   ```
 
-```bash
-mv src/app/app.server.module.ts src/app/app.module.server.ts
-```
+1. In `server.ts`, remove the following imports from the top of the file:
 
-### `src/main.server.ts`
+   ```ts
+   import 'zone.js/node';
+   
+   import { ngExpressEngine as engine } from - '@spartacus/setup/ssr';
+   import { NgExpressEngineDecorator } from - '@spartacus/setup/ssr';
+   import * as express from 'express';
+   import { join } from 'path';
+   
+   import { AppServerModule } from './src/main.- server';
+   import { APP_BASE_HREF } from '@angular/common';
+   import { existsSync } from 'fs';
+   ```
 
-1. Change the the export path of the `AppServerModule` from `./app/app.server.module'` to `./app/app.module.server'`. 
-2. And export `AppServerModule` using `as default`.
+1. In `server.ts`, add the following imports to the top of the file:
 
-```diff
-- export { AppServerModule } from './app/app.server.module';
-+ export { AppServerModule as default } from './app/app.module.server';
-```
+   ```ts
+   import { APP_BASE_HREF } from '@angular/common';
+   import {
+     NgExpressEngineDecorator,
+     ngExpressEngine as engine,
+   } from '@spartacus/setup/ssr';
+   import express from 'express';
+   import { dirname, join, resolve } from 'node:path';
+   import { fileURLToPath } from 'node:url';
+   import AppServerModule from './src/main.server';
+   ```
 
-### `server.ts`
+1. In `server.ts`, remove the following constants:
 
-1. Change the imports in the top of the file according to the following diff:
+   ```ts
+   const distFolder = join(process.cwd(), 'dist/YOUR-APP-NAME/browser');
+   const indexHtml = existsSync(join(distFolder, 'index.original.html'))
+     ? 'index.original.html'
+     : 'index';
+   ```
 
-```diff
-- import 'zone.js/node';
+1. In `server.ts`, add the following constants:
 
-- import { ngExpressEngine as engine } from - '@spartacus/setup/ssr';
-- import { NgExpressEngineDecorator } from - '@spartacus/setup/ssr';
-- import * as express from 'express';
-- import { join } from 'path';
+   ```ts
+   const serverDistFolder = dirname(fileURLToPath(import.meta.url));
+   const browserDistFolder = resolve(serverDistFolder, '../browser');
+   const indexHtml = join(browserDistFolder, 'index.html');
+   ```
```

Comment:
Instead of "remove A" and "add B", I'd like to emphasise that we should "_replace_ A with B". 
In other words, it is important _where_ we add B -> in the exact place where A was.

Reply:
Makes sense! The steps have been adjusted accordingly 🙂 👍